scopes/scopes.go

```diff
@@ -0,0 +1,94 @@
+package scopes
+
+import (
+	"strings"
+)
+
+type (
+	Given    []string
+	Required [][]string
```

Comment:
So I would call these scopesets:
```go
type (
  ScopeSet []string
  ScopeSets []ScopeSet
)
```

The idea is often that you have a scope-set and is required to satisfy one of the _required_ scope-sets.